core/agents/task_completer.py

```diff
@@ -25,5 +26,8 @@ async def run(self) -> AgentResponse:
             self.current_state.get_source_index(source),
             tasks,
         )
+        await telemetry.trace_code_event(
+            "task-end", {"task-num": current_task_index1, "num-iterations": len(self.current_state.iterations)}
```

Comment:
we should either update how trace_code_event works or add data to each trace. Here are some suggestions of what data might be useful:
- number of epics (is it the initial app or feature?)
- number of all tasks (is it the last task? Could be that we have most iterations on the last task because everyone clicks "continue" until it is the last task)
- some data so we can find and analyze the db